### ISSUE TITLE
Implement shim to enable skipping ci on doc only updates while still requiring checks to pass 

### ISSUE THREAD

**Issue by zachariahmiller** [defenseunicorns/uds-package-mattermost#25]
### Describe what should be investigated or refactored
We currently ignore certain files in this repos workflows, such as if README.md is updated. This conflicts with required checks for merge into main. A solution to that is to implement functionality similar to here https://github.com/defenseunicorns/zarf/blob/main/.github/workflows/test-e2e-shim.yml

### Links to any relevant code
(optional) i.e. - https://github.com/defenseunicorns/uds-package-mattermost/blob/main/README.md?plain=1#L1

### Additional context
See discussion [here](https://github.com/defenseunicorns/uds-package-mattermost/pull/23#issuecomment-1948922395)
Also see PR blocked by this not existing [here](https://github.com/defenseunicorns/uds-package-mattermost/pull/23)
# For more info see below:
https://github.com/orgs/community/discussions/54877
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/troubleshooting-required-status-checks#handling-skipped-but-required-checks
